nautobot/extras/tests/test_views.py

```diff
@@ -1800,6 +1801,45 @@ def test_run_job_with_sensitive_variables_for_future(self, _):
             content = extract_page_body(response.content.decode(response.charset))
             self.assertIn("Unable to schedule job: Job may have sensitive input variables.", content)
 
+    @mock.patch("nautobot.extras.views.get_worker_count", return_value=1)
+    def test_run_job_with_sensitive_variables_and_requires_approval(self, _):
+        self.add_permissions("extras.run_job")
+        self.add_permissions("extras.view_scheduledjob")
+
+        self.test_pass.has_sensitive_variables = True
+        self.test_pass.approval_required = True
+        self.test_pass.save()
+
+        data = {
+            "_schedule_type": "immediately",
+        }
+        for run_url in self.run_urls:
+            # Assert warning message shows in get
+            response = self.client.get(run_url)
+            content = extract_page_body(response.content.decode(response.charset))
+            self.assertIn(
+                "This job is flagged as possibly having sensitive variables but is also flagged as requiring approval.",
+                content,
+            )
+
+            # Assert run button is disabled
+            search = re.search(
+                r'<button type="submit" name="_run" id="id__run" class="btn btn-primary"\s+disabled="disabled"', content
```

Comment:
```suggestion
                r'<button[^>]+id="id__run"[^>]+disabled="disabled"', content
```
Should be a but more durable for any CSS changes.